
diff --git a/switcheroo-control.service.in b/switcheroo-control.service.in
index 9bb6939..159a907 100644
--- a/switcheroo-control.service.in
+++ b/switcheroo-control.service.in
@@ -1,21 +1,10 @@
-[Unit]
-Description=Switcheroo Control Proxy service
-Before=multi-user.target display-manager.service alsa-restore.service alsa-state.service
-
-[Service]
-Type=dbus
-BusName=net.hadess.SwitcherooControl
-ExecStart=@libexecdir@/switcheroo-control
-
-# Lockdown
-ProtectSystem=strict
-ProtectControlGroups=true
-ProtectHome=true
-ProtectKernelModules=true
-PrivateTmp=true
-RestrictAddressFamilies=AF_UNIX AF_LOCAL AF_NETLINK
-MemoryDenyWriteExecute=true
-RestrictRealtime=true
-
-[Install]
-WantedBy=graphical.target
+#  SPDX-License-Identifier: GPL-3.0-or-later
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU General Public License version 3 as published by
+# the Free Software Foundation.
+ 
+[D-BUS Service]
+Name=net.hadess.SwitcherooControl
+Exec=@libexecdir@/switcheroo-control &
+User=root
