From 9504806dc3535091535a66e30b9ff02dcb539464 Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Mon, 30 Oct 2023 02:21:04 +0000
Subject: [PATCH 1/4] Update meson.build

---
 session/meson.build | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/session/meson.build b/session/meson.build
index 221560a..2390fc9 100644
--- a/session/meson.build
+++ b/session/meson.build
@@ -127,17 +127,11 @@ if get_option('detect-program-prefixes') == true
   orca_prefix = orca.path().split('/bin')[0]
 
   autostarts = {
-    'gnome-keyring-pkcs11': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-pkcs11.desktop'),
-    'gnome-keyring-secrets': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-secrets.desktop'),
-    'gnome-keyring-ssh': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-ssh.desktop'),
     'onboard-autostart': join_paths(onboard_prefix, 'etc/xdg/autostart', 'onboard-autostart.desktop'),
     'orca-autostart': join_paths(orca_prefix, 'etc/xdg/autostart', 'orca-autostart.desktop'),
   }
 else
   autostarts = {
-    'gnome-keyring-pkcs11': join_paths(autostartdir, 'gnome-keyring-pkcs11.desktop'),
-    'gnome-keyring-secrets': join_paths(autostartdir, 'gnome-keyring-secrets.desktop'),
-    'gnome-keyring-ssh': join_paths(autostartdir, 'gnome-keyring-ssh.desktop'),
     'onboard-autostart': join_paths(autostartdir, 'onboard-autostart.desktop'),
     'orca-autostart': join_paths(autostartdir, 'orca-autostart.desktop'),
   }

From 71ce7e696e9cb2ee1017372c3417980ac109b267 Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Mon, 30 Oct 2023 02:39:17 +0000
Subject: [PATCH 2/4] Update meson.build

---
 session/meson.build | 2 --
 1 file changed, 2 deletions(-)

diff --git a/session/meson.build b/session/meson.build
index 2390fc9..dcda2a6 100644
--- a/session/meson.build
+++ b/session/meson.build
@@ -116,13 +116,11 @@ foreach component : gsd_components
   )
 endforeach
 
-gnome_keyring = find_program('gnome-keyring-daemon')
 onboard = find_program('onboard')
 orca = find_program('orca')
 
 if get_option('detect-program-prefixes') == true
   # TODO: use fs module in meson 0.53.0
-  gnome_keyring_prefix = gnome_keyring.path().split('/bin')[0]
   onboard_prefix = onboard.path().split('/bin')[0]
   orca_prefix = orca.path().split('/bin')[0]
 

From 3f45abbc1825b174735de662fe0c4590732407bf Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Mon, 30 Oct 2023 02:48:02 +0000
Subject: [PATCH 3/4] Update meson.build

---
 meson.build | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meson.build b/meson.build
index cf95302..06d1145 100644
--- a/meson.build
+++ b/meson.build
@@ -11,3 +11,7 @@ endif
 
 subdir('session')
 subdir('xsessions')
+
+ if get_option('wayland')
+     subdir('wayland-sessions')
+ endif

From 78921d4e805cf9eedd06f8f1f310bab5441ad9a3 Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Mon, 30 Oct 2023 02:49:08 +0000
Subject: [PATCH 4/4] Update meson_options.txt

---
 meson_options.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/meson_options.txt b/meson_options.txt
index 9734566..32ce3e8 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -6,3 +6,6 @@ option('fallback-session', type : 'string', value: 'ubuntu',
 
 option('detect-program-prefixes', type: 'boolean', value: false,
        description : 'This is needed for platforms that install packages into individual prefixes (like Guix System or NixOS')
+
+ option('wayland', type: 'boolean', value: false,
+        description: 'Install Wayland session')
