From 9504806dc3535091535a66e30b9ff02dcb539464 Mon Sep 17 00:00:00 2001
From: Nathaniel Russell <46272571+nater1983@users.noreply.github.com>
Date: Mon, 30 Oct 2023 02:21:04 +0000
Subject: [PATCH] Update meson.build

---
 session/meson.build | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/session/meson.build b/session/meson.build
index 221560a..2390fc9 100644
--- a/session/meson.build
+++ b/session/meson.build
@@ -127,17 +127,11 @@ if get_option('detect-program-prefixes') == true
   orca_prefix = orca.path().split('/bin')[0]
 
   autostarts = {
-    'gnome-keyring-pkcs11': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-pkcs11.desktop'),
-    'gnome-keyring-secrets': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-secrets.desktop'),
-    'gnome-keyring-ssh': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-ssh.desktop'),
     'onboard-autostart': join_paths(onboard_prefix, 'etc/xdg/autostart', 'onboard-autostart.desktop'),
     'orca-autostart': join_paths(orca_prefix, 'etc/xdg/autostart', 'orca-autostart.desktop'),
   }
 else
   autostarts = {
-    'gnome-keyring-pkcs11': join_paths(autostartdir, 'gnome-keyring-pkcs11.desktop'),
-    'gnome-keyring-secrets': join_paths(autostartdir, 'gnome-keyring-secrets.desktop'),
-    'gnome-keyring-ssh': join_paths(autostartdir, 'gnome-keyring-ssh.desktop'),
     'onboard-autostart': join_paths(autostartdir, 'onboard-autostart.desktop'),
     'orca-autostart': join_paths(autostartdir, 'orca-autostart.desktop'),
   }
