From adcbb4bcf21087f23900e2d112bbbe19e674d23d Mon Sep 17 00:00:00 2001
From: lenemter <lenemter@gmail.com>
Date: Tue, 20 Aug 2024 23:11:39 +0900
Subject: [PATCH 3/3] Revert one change

---
 src/ScreenshotManager.vala | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/ScreenshotManager.vala b/src/ScreenshotManager.vala
index 3223e3036..ec74631b6 100644
--- a/src/ScreenshotManager.vala
+++ b/src/ScreenshotManager.vala
@@ -361,8 +361,7 @@ namespace Gala {
 
             try {
                 unowned var selection = wm.get_display ().get_selection ();
-                var bytes = new GLib.Bytes.take (buffer);
-                var source = new Meta.SelectionSourceMemory ("image/png", bytes);
+                var source = new Meta.SelectionSourceMemory ("image/png", new GLib.Bytes.take (buffer));
                 selection.set_owner (Meta.SelectionType.SELECTION_CLIPBOARD, source);
             } catch (Error e) {
                 warning ("Could not save screenshot to clipboard: failed to create new Meta.SelectionSourceMemory: %s", e.message);
