diff --git a/data/autostart.desktop b/data/autostart.desktop
deleted file mode 100644
index 08949ac2..00000000
--- a/data/autostart.desktop
+++ /dev/null
@@ -1,14 +0,0 @@
-[Desktop Entry]
-Name=Wingpanel
-Comment=Top panel that holds indicators and the applications menu
-Exec=io.elementary.wingpanel
-Icon=io.elementary.wingpanel
-Terminal=false
-Type=Application
-Categories=Utility;
-NoDisplay=true
-X-GNOME-Autostart-Notify=false
-X-GNOME-AutoRestart=true
-X-GNOME-Autostart-enabled=true
-X-GNOME-Autostart-Phase=Panel
-OnlyShowIn=Pantheon;
diff --git a/data/meson.build b/data/meson.build
index 87fda003..a6a574b6 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -4,12 +4,6 @@ install_data(
     install_dir: join_paths(get_option('datadir'), 'icons', 'hicolor', 'scalable', 'apps')
 )
 
-install_data(
-    'autostart.desktop',
-    rename: meson.project_name() + '.desktop',
-    install_dir: join_paths(get_option('sysconfdir'), 'xdg', 'autostart')
-)
-
 i18n.merge_file(
     input: 'wingpanel.desktop.in',
     output: meson.project_name() + '.desktop',
