diff --git a/meson.build b/meson.build
index 1520965..f6e875d 100644
--- a/meson.build
+++ b/meson.build
@@ -19,7 +19,6 @@ granite_dep = dependency('granite-7', version: '>=7.7.0')
 gtk_dep = dependency('gtk4')
 gtk_wayland_dep = dependency('gtk4-wayland')
 handy_dep = dependency('libadwaita-1', version: '>=1.0')
-packagekit_dep = dependency('packagekit-glib2')
 wingpanel_dep = dependency('wingpanel-9')
 wingpanel_indicatorsdir = wingpanel_dep.get_pkgconfig_variable('indicatorsdir', define_variable: ['libdir', libdir])
 wl_client_dep = dependency('wayland-client')
diff --git a/src/Widgets/EndSessionDialog.vala b/src/Widgets/EndSessionDialog.vala
index c4c7173..45a422a 100644
--- a/src/Widgets/EndSessionDialog.vala
+++ b/src/Widgets/EndSessionDialog.vala
@@ -57,7 +57,6 @@ public class QuickSettings.EndSessionDialog : Granite.MessageDialog {
         if (dialog_type == EndSessionDialogType.RESTART) {
             var confirm_restart = (Gtk.Button) add_button (_("Restart"), 1);
             confirm_restart.clicked.connect (() => {
-                set_offline_trigger (REBOOT); // This will just do nothing if no updates are available
                 reboot ();
                 destroy ();
             });
@@ -68,24 +67,6 @@ public class QuickSettings.EndSessionDialog : Granite.MessageDialog {
         var confirm = (Gtk.Button) add_button (button_text, Gtk.ResponseType.ACCEPT);
         confirm.add_css_class (Granite.CssClass.DESTRUCTIVE);
 
-        if (dialog_type != LOGOUT) {
-            bool has_prepared_updates = false;
-            try {
-                has_prepared_updates = Pk.offline_get_prepared_ids ().length > 0;
-            } catch (Error e) {
-                warning ("Failed to check for prepared updates, assuming no: %s", e.message);
-            }
-
-            if (has_prepared_updates) {
-                updates_check_button = new Gtk.CheckButton () {
-                    active = true,
-                    label = _("Install pending system updates"),
-                };
-
-                custom_bin.append (updates_check_button);
-            }
-        }
-
         cancel.grab_focus ();
 
         var cancel_action = new SimpleAction ("cancel", null);
@@ -109,11 +90,7 @@ public class QuickSettings.EndSessionDialog : Granite.MessageDialog {
 
         confirm.clicked.connect (() => {
             if (dialog_type == EndSessionDialogType.RESTART || dialog_type == EndSessionDialogType.SHUTDOWN) {
-                if (set_offline_trigger (POWER_OFF)) {
-                    reboot ();
-                } else {
                     shutdown ();
-                }
             } else {
                 logout ();
             }
@@ -124,31 +101,6 @@ public class QuickSettings.EndSessionDialog : Granite.MessageDialog {
         ((Gtk.Widget) this).realize.connect (() => Idle.add_once (() => init_wl ()));
     }
 
-    private bool set_offline_trigger (Pk.OfflineAction action) {
-        if (updates_check_button == null) {
-            return false;
-        }
-
-        if (updates_check_button.active) {
-            try {
-                Pk.offline_trigger (action);
-                return true;
-            } catch (Error e) {
-                critical ("Failed to set offline trigger for updates: %s", e.message);
-            }
-        } else {
-            try {
-                if (Pk.offline_get_action () != UNSET) {
-                    Pk.offline_cancel ();
-                }
-            } catch (Error e) {
-                critical ("Failed to check/cancel offline trigger for updates: %s", e.message);
-            }
-        }
-
-        return false;
-    }
-
     public void registry_handle_global (Wl.Registry wl_registry, uint32 name, string @interface, uint32 version) {
         if (@interface == "io_elementary_pantheon_shell_v1") {
             var desktop_shell = wl_registry.bind<PantheonDesktop.Shell> (name, ref PantheonDesktop.Shell.iface, uint32.min (version, 1));
diff --git a/src/meson.build b/src/meson.build
index 8979761..ece2cbe 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -43,7 +43,6 @@ shared_module(
         gtk_wayland_dep,
         handy_dep,
         wingpanel_dep,
-        packagekit_dep,
         pantheon_desktop_shell_dep,
         portal_dep,
         wl_client_dep,
