From 2c7a4f8b177d1000790af67a648cb364ff0c31a5 Mon Sep 17 00:00:00 2001
From: Muhammed Efe Cetin <efectn@protonmail.com>
Date: Mon, 17 Jul 2023 18:36:03 +0300
Subject: [PATCH 1/1] Fix hardcoded GTK theming

---
 src/WindowManager.vala | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/WindowManager.vala b/src/WindowManager.vala
index ecff8e86..dc78b29c 100644
--- a/src/WindowManager.vala
+++ b/src/WindowManager.vala
@@ -85,7 +85,7 @@ namespace Gala {
          */
         private Zoom? zoom = null;
 
-        private AccentColorManager accent_color_manager;
+        // private AccentColorManager accent_color_manager;
 
         private Clutter.Actor? tile_preview;
 
@@ -302,7 +302,7 @@ namespace Gala {
                 unowned string[] _args = args;
                 Gtk.init (ref _args);
 
-                accent_color_manager = new AccentColorManager ();
+                // accent_color_manager = new AccentColorManager ();
 
                 // initialize plugins and add default components if no plugin overrides them
                 var plugin_manager = PluginManager.get_default ();
-- 
2.41.0

