From e15ec75ec2ee4cca5e9d360d992714983c8a1df3 Mon Sep 17 00:00:00 2001
From: Marius Meisenzahl <meisenzahl@users.noreply.github.com>
Date: Tue, 20 Aug 2024 12:17:23 +0200
Subject: [PATCH 1/2] meson: Remove onboard

---
 session/meson.build | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/session/meson.build b/session/meson.build
index 501e836..fdd4e34 100644
--- a/session/meson.build
+++ b/session/meson.build
@@ -67,20 +67,17 @@ foreach component : gsd_components
 endforeach
 
 gnome_keyring = find_program('gnome-keyring-daemon')
-onboard = find_program('onboard')
 orca = find_program('orca')
 
 if get_option('detect-program-prefixes') == true
   # TODO: use fs module in meson 0.53.0
   gnome_keyring_prefix = gnome_keyring.path().split('/bin')[0]
-  onboard_prefix = onboard.path().split('/bin')[0]
   orca_prefix = orca.path().split('/bin')[0]
 
   autostarts = {
     'gnome-keyring-pkcs11': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-pkcs11.desktop'),
     'gnome-keyring-secrets': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-secrets.desktop'),
     'gnome-keyring-ssh': join_paths(gnome_keyring_prefix, 'etc/xdg/autostart', 'gnome-keyring-ssh.desktop'),
-    'onboard-autostart': join_paths(onboard_prefix, 'etc/xdg/autostart', 'onboard-autostart.desktop'),
     'orca-autostart': join_paths(orca_prefix, 'etc/xdg/autostart', 'orca-autostart.desktop'),
   }
 else
@@ -88,7 +85,6 @@ else
     'gnome-keyring-pkcs11': join_paths(autostartdir, 'gnome-keyring-pkcs11.desktop'),
     'gnome-keyring-secrets': join_paths(autostartdir, 'gnome-keyring-secrets.desktop'),
     'gnome-keyring-ssh': join_paths(autostartdir, 'gnome-keyring-ssh.desktop'),
-    'onboard-autostart': join_paths(autostartdir, 'onboard-autostart.desktop'),
     'orca-autostart': join_paths(autostartdir, 'orca-autostart.desktop'),
   }
 endif

From e60bfcf0b8154cfc7d1a8744b96dc184c30d4dee Mon Sep 17 00:00:00 2001
From: Marius Meisenzahl <meisenzahl@users.noreply.github.com>
Date: Tue, 20 Aug 2024 12:17:37 +0200
Subject: [PATCH 2/2] GitHub Actions: Do not install onboard

---
 .github/workflows/main.yml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/.github/workflows/main.yml b/.github/workflows/main.yml
index 2d54de7..3b6bbfe 100644
--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -27,7 +27,7 @@ jobs:
     - name: Install Dependencies
       run: |
         apt update
-        apt install -y meson pkg-config gnome-settings-daemon gnome-settings-daemon-dev onboard gnome-keyring orca
+        apt install -y meson pkg-config gnome-settings-daemon gnome-settings-daemon-dev gnome-keyring orca
 
     - name: Build
       run: |
